command/exclude.go

```diff
@@ -0,0 +1,41 @@
+package command
+
+import (
+	"regexp"
+	"strings"
+)
+
+func wildCardToRegexp(pattern string) string {
+	patternRegex := regexp.QuoteMeta(pattern)
+	patternRegex = strings.Replace(patternRegex, "\\?", ".", -1)
+	patternRegex = strings.Replace(patternRegex, "\\*", ".*", -1)
+	return patternRegex
```

Comment:
For given pattern `*.txt`, regex produced here also matches `some-file.txt.extension` which I think is not an expected behavior.

Reply:
what if we add $ to end of `patternRegex`. That might solve the problem? What do you think?

Reply:
Works for case above and current tests cases, we should add this one to tests as well. 
But it'll basically work as `.*<pattern>$` without `^` anchor. I tried prepending `^` to make it more restrictive but that breaks cases with local directory.

Reply:
I added simple test case for `ls_test.go` with a file which has extension `.txt.extension`. After adding `$` to regex patterns, now it will not exclude the file with  `.txt.extension`.

Reply:
Found case below that would ignore unexpected files due to pattern acting like `.*<pattern>$` without `^`.

Have following remote and local files,
```
$ s5cmd ls 's3://bucket/prefix/nested-dir/*'
2021/08/04 06:49:41                 0 file.txt
2021/08/04 06:49:41                 0 some-dir/file.txt
2021/08/04 06:49:41                 0 some-dir/some-dir/file.txt
2021/08/04 06:49:41                 0 some-dir/some-dir/some-dir/file.txt


$ tree nested-dir
nested-dir
├── file.txt
└── some-dir
    ├── file.txt
    └── some-dir
        ├── file.txt
        └── some-dir
            └── file.txt

3 directories, 4 files
```

Running `s5cmd --exclude 'some-dir/some-dir/file.txt' ls` on both of them also ignores `some-dir/some-dir/some-dir/file.txt`;

```
s5cmd ls --exclude 'some-dir/some-dir/file.txt' 'nested-dir/'
2021/08/04 09:41:56                 0 file.txt
2021/08/04 09:42:01                 0 some-dir/file.txt


s5cmd ls --exclude 'some-dir/some-dir/file.txt' 's3://bucket/prefix/nested/*'
2021/08/04 06:49:41                 0 file.txt
2021/08/04 06:49:41                 0 some-dir/file.txt

```

Reply:
it should work with the new commit.